### DISCUSSION TITLE
How to generate 3D microstructures with periodicity ?

### DISCUSSION THREAD

**Discussion by joshuajaistein** [kip-hart/MicroStructPy#57] [Q&A] [answered]
I want to generate periodic , 3 dimentional crystals.

**Comment by kip-hart** [comment, marked as answer]
Hi Joshua,

That would be a great feature, however periodicity is currently not possible with MicroStructPy. [Voro++](http://math.lbl.gov/voro++/about.html) and [pyvoro](https://github.com/joe-jordan/pyvoro) can both create periodic polyhedral structures. 

For the meshing, it doesn't _seem_ like [MeshPy](https://documen.tician.de/meshpy/index.html) supports periodicity. It looks like [gmsh/pygmsh](https://github.com/nschloe/pygmsh/issues/193) supports periodic output.

So it looks like periodic meshes could be produced if using gmsh. Would you be willing to help add this feature? It would be a bigger effort than I can currently manage in the near term.

Best,
Kip
